crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -312,6 +312,43 @@ def create_proposal
     base
   end
 
+  def prepare_nodes_for_crowbar_upgrade
+    proposal = Proposal.find_by(barclamp: "crowbar", name: "default")
+
+    # To all nodes, add a new role which prepares them for the upgrade
+    nodes_to_upgrade = []
+    not_ready_for_upgrade = []
+    all_nodes = NodeObject.all
+    all_nodes.each do |n|
+      not_ready_for_upgrade.push n.name if !n.admin? && !%w(ready crowbar_upgrade).include?(n.state)
```

Comment:
Not sure I think this style is just not tracked by hound. At least we should fix it now the other commit is from the time where we just merged the stuff.